src/github.com/getlantern/lantern-mobile/app/src/main/java/org/getlantern/lantern/fragment/FeedFragment.java

```diff
@@ -31,29 +31,22 @@
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        this.mFeedItems = new ArrayList<FeedItem>();
+        this.adapter = new FeedAdapter(getActivity(), mFeedItems);
 
         Bundle bundle = getArguments();
-        this.feedName = bundle.getString("name");
-        this.mFeedItems = Collections.synchronizedList(new ArrayList<FeedItem>());
+        if (bundle != null) {
+            this.feedName = bundle.getString("name");
+        }
     }
 
     public String getFeedName() {
         return feedName;
     }
 
-    public void NotifyDataSetChanged(final List<FeedItem> items) {
-        getActivity().runOnUiThread(new Runnable() {
-            public void run() {
-                mFeedItems.clear();
-                mFeedItems.addAll(items);
-                Log.d(TAG, String.format("Feed %s has %d items", feedName, 
-                            mFeedItems.size()));
-                if (adapter != null) {
-                    adapter.notifyDataSetChanged(); 
-                }
-            }
-        });
-    }   
+    public void setFeedName(String feedName) {
+        this.feedName = feedName;
```

Comment:
This is called automatically by the JSON parser or something @atavism is that right?